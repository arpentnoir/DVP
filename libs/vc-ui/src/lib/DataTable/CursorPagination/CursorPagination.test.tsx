import { TablePaginationProps } from '@mui/material';
import { fireEvent, render } from '@testing-library/react';
import {
  CursorPagination,
  nextPageTestId,
  PaginationControls,
  prevPageTestId,
} from './CursorPagination';

const mockHandleNextPage = jest.fn();
const mockHandlePrevPage = jest.fn();
const mockNextCursor = null;
const mockPrevCursor = null;
const limit = 50;

const mockedCursorPaginationProps = {
  handleNextPage: mockHandleNextPage,
  handlePrevPage: mockHandlePrevPage,
  nextCursor: mockNextCursor,
  prevCursor: mockPrevCursor,
  limit,
} as unknown as PaginationControls & TablePaginationProps;

describe('CursorPagination', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    const { baseElement } = render(
      <CursorPagination {...mockedCursorPaginationProps} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should disable both buttons if next/prev cursor are null', () => {
    const { getByTestId } = render(
      <CursorPagination {...mockedCursorPaginationProps} />
    );

    expect(getByTestId(prevPageTestId)).toHaveAttribute('disabled');
    expect(getByTestId(nextPageTestId)).toHaveAttribute('disabled');
  });

  it('should enable next button and disable prev button', () => {
    const { getByTestId } = render(
      <CursorPagination
        {...mockedCursorPaginationProps}
        nextCursor={'nextCursor'}
      />
    );

    expect(getByTestId(prevPageTestId)).toHaveAttribute('disabled');
    expect(getByTestId(nextPageTestId)).not.toHaveAttribute('disabled');
  });

  it('should enable prev button and disable next button', () => {
    const { getByTestId } = render(
      <CursorPagination
        {...mockedCursorPaginationProps}
        prevCursor={'prevCursor'}
      />
    );

    expect(getByTestId(prevPageTestId)).not.toHaveAttribute('disabled');
    expect(getByTestId(nextPageTestId)).toHaveAttribute('disabled');
  });

  it('should enable next/prev button', () => {
    const { getByTestId } = render(
      <CursorPagination
        {...mockedCursorPaginationProps}
        nextCursor={'nextCursor'}
        prevCursor={'prevCursor'}
      />
    );

    expect(getByTestId(prevPageTestId)).not.toHaveAttribute('disabled');
    expect(getByTestId(nextPageTestId)).not.toHaveAttribute('disabled');
  });

  it('should call the handleNextPage function when next button is clicked', () => {
    const { getByTestId } = render(
      <CursorPagination
        {...mockedCursorPaginationProps}
        nextCursor={'nextCursor'}
      />
    );

    const nextPageButton = getByTestId(nextPageTestId);
    fireEvent.click(nextPageButton);

    expect(mockHandleNextPage).toHaveBeenCalledTimes(1);
    expect(mockHandlePrevPage).not.toHaveBeenCalled();
  });

  it('should call the handlePrevPage function when prev button is clicked', () => {
    const { getByTestId } = render(
      <CursorPagination
        {...mockedCursorPaginationProps}
        prevCursor={'prevCursor'}
      />
    );

    const prevPageButton = getByTestId(prevPageTestId);
    fireEvent.click(prevPageButton);

    expect(mockHandlePrevPage).toHaveBeenCalledTimes(1);
    expect(mockHandleNextPage).not.toHaveBeenCalled();
  });
});
