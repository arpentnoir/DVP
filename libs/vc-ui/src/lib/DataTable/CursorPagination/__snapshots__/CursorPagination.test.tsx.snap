// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CursorPagination should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiTablePagination-root css-1gl81ut-MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          1â€“1 of 1
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            data-testid="button:cursor-pagination-prev"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            data-testid="button:cursor-pagination-next"
            disabled=""
            tabindex="-1"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
